homeassistant/components/nexia/climate.py

```diff
@@ -0,0 +1,396 @@
+        # temp range.
+        if hold and mode == OPERATION_MODE_AUTO:
+            return HVAC_MODE_HEAT_COOL
+
+        return NEXIA_TO_HA_HVAC_MODE_MAP[mode]
+
+    @property
+    def hvac_modes(self):
+        """List of HVAC available modes."""
+        return [
+            HVAC_MODE_OFF,
+            HVAC_MODE_AUTO,
+            HVAC_MODE_HEAT_COOL,
+            HVAC_MODE_HEAT,
+            HVAC_MODE_COOL,
+        ]
+
+    def set_temperature(self, **kwargs):
+        """Set target temperature."""
+        new_heat_temp = kwargs.get(ATTR_TARGET_TEMP_LOW, None)
+        new_cool_temp = kwargs.get(ATTR_TARGET_TEMP_HIGH, None)
+        set_temp = kwargs.get(ATTR_TEMPERATURE, None)
+
+        deadband = self.thermostat.get_deadband()
+        cur_cool_temp = self._device.get_cooling_setpoint()
+        cur_heat_temp = self._device.get_heating_setpoint()
+        (min_temp, max_temp) = self.thermostat.get_setpoint_limits()
+
+        # Check that we're not going to hit any minimum or maximum values
+        if new_heat_temp and new_heat_temp + deadband > max_temp:
+            new_heat_temp = max_temp - deadband
+        if new_cool_temp and new_cool_temp - deadband < min_temp:
+            new_cool_temp = min_temp + deadband
+
+        # Check that we're within the deadband range, fix it if we're not
+        if new_heat_temp and new_heat_temp != cur_heat_temp:
+            if new_cool_temp - new_heat_temp < deadband:
+                new_cool_temp = new_heat_temp + deadband
+        if new_cool_temp and new_cool_temp != cur_cool_temp:
+            if new_cool_temp - new_heat_temp < deadband:
+                new_heat_temp = new_cool_temp - deadband
+
+        self._device.set_heat_cool_temp(
+            heat_temperature=new_heat_temp,
+            cool_temperature=new_cool_temp,
+            set_temperature=set_temp,
+        )
+        self.schedule_update_ha_state()
+
+    @property
+    def is_aux_heat(self):
+        """Emergency heat state."""
+        return self.thermostat.is_emergency_heat_active()
+
+    @property
+    def device_info(self):
+        """Return the device_info of the device."""
+        return {
+            "identifiers": {(DOMAIN, self._device.zone_id)},
+            "name": self._device.get_name(),
+            "model": self.thermostat.get_model(),
+            "sw_version": self.thermostat.get_firmware(),
+            "manufacturer": MANUFACTURER,
+            "via_device": (DOMAIN, self.thermostat.thermostat_id),
+        }
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        data = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_ZONE_STATUS: self._device.get_status(),
+        }
+
+        if self._has_emergency_heat:
+            aux_heat_value = "off"
+            if self.thermostat.is_emergency_heat_active():
+                aux_heat_value = "on"
+            data.update({ATTR_AUX_HEAT: aux_heat_value})
```

Comment:
I think we can drop this too.

https://github.com/home-assistant/core/blob/ff2367fffb3329fb7bd5f00412ffd7daa4e77fdd/homeassistant/components/climate/__init__.py#L259-L260

Reply:
This one is supplemental heating source that is used for emergencies in in northern climates (below 35 degrees F) so you don't freeze.  Since its not represented by any other state, I think its ok to stay

Reply:
In case its an interesting topic: https://www.hannabery.com/emergency-heat-setting-when-to-use-is-it-more-expensive-red-light-on.shtml

Reply:
It is represented by the state attribute aux heat that I linked to.

Reply:
It absolutely is.  I completely misread your comment.  Removed!